File: in_Java/0342 Power of Four.java
class Solution {
    public boolean isPowerOfFour(int num) {
        if(num < 0) return false;
        int count_zero = 0;
        int count_one = 0;
        while(num != 0){
            if(num % 2 == 0) count_zero++;
            else count_one++;
            num/=2;
        }
        if(count_one == 1 && count_zero % 2 == 0) return true;
        else return false;
    }
}

File: in_Python/0342 Power of Four.py
class Solution(object):
    def isPowerOfFour(self, num):
        """
        :type num: int
        :rtype: bool
        """
        sumz = 0
        n = num
        while n != 0 and n != 1:
            x = n % 2
            if x != 0:
                return False
            sumz += 1
            n /= 2
        if n == 0:
            return False
        return sumz % 2 == 0
        